tests/cmip6/test_cmip6.py

```diff
@@ -119,3 +121,21 @@ def test_load_esmcol_remote():
         'https://raw.githubusercontent.com/NCAR/intake-esm-datastore/master/catalogs/pangeo-cmip6.json'
     )
     assert isinstance(col.df, pd.DataFrame)
+
+
+def test_serialize():
+    with TemporaryDirectory() as local_store:
+        col = intake.open_esm_datastore(
+            'https://raw.githubusercontent.com/NCAR/intake-esm-datastore/master/catalogs/pangeo-cmip6.json'
+        )
+        col_subset = col.search(
+            source_id='BCC-ESM1', grid_label='gn', table_id='Amon', experiment_id='historical'
+        )
+
+        name = 'cmip6_bcc_esm1'
+        col_subset.serialize(name=name, directory=local_store)
+
+        col = intake.open_esm_datastore(f'{local_store}/cmip6_bcc_esm1.json')
+        pd.testing.assert_frame_equal(col_subset.df.reset_index(drop=True), col.df)
```

Comment:
the `reset_index` piece is worrying here. Why is this needed?

Reply:
> the reset_index piece is worrying here. Why is this needed?

`col.search()` returns dataframe with original indices. When saving the subset, I am not keeping these original indices. As a result, `pd.testing.assert_frame_equal()` fails when used without `reset_index()`:


```python
In [3]: col = intake.open_esm_datastore("https://raw.githubusercontent.com/NCAR/intake-esm-datastore/master/catalogs/pangeo-cmi
   ...: p6.json")                                                                                                              

In [4]: col.df.head()                                                                                                          
Out[4]: 
  activity_id institution_id source_id  ... grid_label                                             zstore dcpp_init_year
0  AerChemMIP            BCC  BCC-ESM1  ...         gn  gs://cmip6/AerChemMIP/BCC/BCC-ESM1/piClim-CH4/...            NaN
1  AerChemMIP            BCC  BCC-ESM1  ...         gn  gs://cmip6/AerChemMIP/BCC/BCC-ESM1/piClim-CH4/...            NaN
2  AerChemMIP            BCC  BCC-ESM1  ...         gn  gs://cmip6/AerChemMIP/BCC/BCC-ESM1/piClim-CH4/...            NaN
3  AerChemMIP            BCC  BCC-ESM1  ...         gn  gs://cmip6/AerChemMIP/BCC/BCC-ESM1/piClim-CH4/...            NaN
4  AerChemMIP            BCC  BCC-ESM1  ...         gn  gs://cmip6/AerChemMIP/BCC/BCC-ESM1/piClim-CH4/...            NaN

[5 rows x 10 columns]

In [5]: col_subset = col.search(source_id="BCC-ESM1", grid_label="gn", table_id="Amon", experiment_id="historical")            

In [6]: col_subset.df.head()                                                                                                   
Out[6]: 
     activity_id institution_id source_id  ... grid_label                                             zstore dcpp_init_year
5416        CMIP            BCC  BCC-ESM1  ...         gn  gs://cmip6/CMIP/BCC/BCC-ESM1/historical/r1i1p1...            NaN
5417        CMIP            BCC  BCC-ESM1  ...         gn  gs://cmip6/CMIP/BCC/BCC-ESM1/historical/r1i1p1...            NaN
5418        CMIP            BCC  BCC-ESM1  ...         gn  gs://cmip6/CMIP/BCC/BCC-ESM1/historical/r1i1p1...            NaN
5419        CMIP            BCC  BCC-ESM1  ...         gn  gs://cmip6/CMIP/BCC/BCC-ESM1/historical/r1i1p1...            NaN
5420        CMIP            BCC  BCC-ESM1  ...         gn  gs://cmip6/CMIP/BCC/BCC-ESM1/historical/r1i1p1...            NaN
```

Reply:
Should we keep these original indices?

Reply:
I think we should either save the original indicies OR have `search()` return reset its own indicies. Ultimately, I think we just want to be internally consistent here.

Reply:
I concur! I added the `reset_index()` step as part of `search()` for internal consistency